src/hooks/useTrades.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react'
+
+import { Trade } from 'api/exchange/ExchangeApi'
+
+import { getTrades } from 'services'
+
+import { appendTrades } from 'reducers-actions/trades'
+
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import useGlobalState from 'hooks/useGlobalState'
+import { CONFIRMATION_BLOCKS } from 'const'
+
+// TODO: (on global state) store trades to local storage
+// TODO: account for reverts
+export function useTrades(): Trade[] {
+  const [
+    {
+      trades: { trades, lastCheckedBlock },
+    },
+    dispatch,
+  ] = useGlobalState()
+  const { userAddress, networkId, blockNumber } = useWalletConnection()
+
+  useEffect(() => {
+    // Check only up to `latest - CONFIRMATION_BLOCKS` to reduce chance of re-orgs
+    const toBlock = blockNumber && blockNumber - CONFIRMATION_BLOCKS
+
+    if (userAddress && networkId && toBlock && (!lastCheckedBlock || lastCheckedBlock < toBlock)) {
+      getTrades({
+        userAddress,
+        networkId,
+        fromBlock: lastCheckedBlock,
+        toBlock,
+      }).then(newTrades => dispatch(appendTrades(newTrades, toBlock)))
```

Comment:
This would dispatch even if `newTrades` is empty, causing reducer run and rerender. Better to opt out if no new trades

Reply:
Or don't change state inside reducer when incoming trades is empty

Reply:
Since we still need to update `toBlock` anyway. So either a new action for `lastCheckedBlock` update only. Or check ifEmpty in reducer.
I prefer new action, like we already do for 'UPDATE_OFFSET` in orders

Reply:
I hoped it would be smart enough.
Alright then, I'll update only the `lastCheckedBlock` when no new trades.